### PR TITLE
feat(v2-eos-cores): <- allow those to work with multi incremerkles

### PR DESCRIPTION
**Pull request by gskapka**
...per title.

The eos cores now store up to X previous incremerkles, allowing for submission material to contains proofs that reference any of those rather than just the chain tip.

This is to help mitigate the race condition that sometimes occurs between the two nodes used for eos cores, one for getting blocks and the other for actions.

If the action is slow to appear on the end point, occasionally the core's incremerkle can have been updated in the meanwhile, moving it beyond the block the action (when it appears) was dispated in. This leaves the submission stuck.

Now however, the cores when asked about their latest block numbers, return an array of options that the eos-syncer may chose from in order to correctly proof an action w/r/t to a block the core knows about.

### CODE REVIEW

File: common/eos/src/eos_incremerkle.rs [hunk 1]
```diff
@@ -0,0 +1,795 @@
...
+    traits::DatabaseInterface,
+    types::{NoneError, Result},
+};
+use derive_more::{Constructor, Deref, DerefMut};
+use eos_chain::Checksum256;
+use serde::{Deserialize, Serialize};
+
+use crate::{EosDbUtils, EosState};
+
+// NOTE: The light client for EOS doesn't not keep blocks - they are too frequent and too numerous
```
Comment (ubordignon) [inline common/eos/src/eos_incremerkle.rs:14]:
```suggestion
// NOTE: The light client for EOS doesn't keep blocks - they are too frequent and too numerous
```

File: common/eos/src/eos_incremerkle.rs [hunk 2]
```diff
@@ -0,0 +1,795 @@
...
+// for efficient use in TEEs.
+//
+// Instead, we provide as the first trusted block from the node which contains all the
```
Comment (ubordignon) [inline common/eos/src/eos_incremerkle.rs:17]:
```suggestion
// Instead, we provide as the first trusted block one from the node which contains all the
```

**Comment by gskapka** [comment]
Awaiting testing on prod before merging.

**Review by ubordignon — DISMISSED** [review]
LGTM!

**Review by ubordignon — APPROVED** [review]
LGTM!

**Pull request merged**